src/main/java/org/kiwiproject/registry/eureka/client/EurekaParser.java

```diff
@@ -67,43 +72,64 @@ private static EurekaInstance buildInstance(Map<String, Object> instanceData) {
+                .secureVipAddress(getStringOrNull(instanceData, "vipAddress"))
+                .app(getStringOrNull(instanceData, "app"))
+                .hostName(getStringOrNull(instanceData, "hostName"))
+                .ipAddr(getStringOrNull(instanceData, "ipAddr"))
+                .status(getStringOrNull(instanceData, "status"))
+                .homePageUrl(getStringOrNull(instanceData, "homePageUrl"))
+                .healthCheckUrl(healthCheckUrl)
+                .statusPageUrl(statusUrl)
+                .port(portMap)
+                .securePort(securePortMap)
+                .adminPort(adminPort)
+                .leaseInfo(leaseInfo)
+                .metadata(metadataMap)
+                .build();
+    }
+
+    private static Map<String, Object> extractLeaseInfo(Map<String, Object> leaseInfoMap) {
         if (nonNull(leaseInfoMap)) {
-            leaseInfo = Map.of(
+            return Map.of(
                     "renewalIntervalInSecs", leaseInfoMap.getOrDefault("renewalIntervalInSecs", 0),
                     "durationInSecs", leaseInfoMap.getOrDefault("durationInSecs", 0),
                     "registrationTimestamp", leaseInfoMap.getOrDefault("registrationTimestamp", 0L),
                     "lastRenewalTimestamp", leaseInfoMap.getOrDefault("lastRenewalTimestamp", 0L),
                     "evictionTimestamp", leaseInfoMap.getOrDefault("evictionTimestamp", 0L),
                     "serviceUpTimestamp", leaseInfoMap.getOrDefault("serviceUpTimestamp", 0L));
-        } else {
-            leaseInfo = Map.of();
         }
 
-        return EurekaInstance.builder()
-                .vipAddress(getString(instanceData, "vipAddress"))
-                .secureVipAddress(getString(instanceData, "vipAddress"))
-                .app(getString(instanceData, "app"))
-                .hostName(getString(instanceData, "hostName"))
-                .ipAddr(getString(instanceData, "ipAddr"))
-                .status(getString(instanceData, "status"))
-                .homePageUrl(getString(instanceData, "homePageUrl"))
-                .healthCheckUrl(getString(instanceData, "healthCheckUrl"))
-                .statusPageUrl(getString(instanceData, "statusUrl"))
-                .port(portMap)
-                .securePort(securePortMap)
-                .leaseInfo(leaseInfo)
-                .metadata(metadataMap)
-                .build();
+        return Map.of();
     }
 
     @SuppressWarnings("ConstantConditions")
-    private static <K> String getString(final Map<? super K, ?> map, final K key) {
+    private static <K> String getStringOrNull(final Map<? super K, ?> map, final K key) {
         var value = map.getOrDefault(key, null);
 
         verify(isNull(value) || value instanceof String, "Value from Map must be a string or null");
 
         return (String) value;
     }
 
+    @VisibleForTesting
+    static int getAdminPort(@Nullable String url) {
+        if (isBlank(url)) {
+            return 0;
```

Comment:
We actually should probably make this -1 since 0 is a "special" port

Reply:
Or maybe not since they seem to prefer 0 :-(

Reply:
OK, leaving as is then.